default/python/universe_generation/specials.py

```diff
@@ -1,17 +1,68 @@
 import random
+from collections import defaultdict
+
 import freeorion as fo
 import statistics
 import universe_tables
+from galaxy_topology import get_systems_within_jumps
+
+# REPEAT_RATE along with calculate_number_of_specials_to_place determines if there are multiple
+# specials in a single location.  There can only be at most 4 specials in a single location.
+# The probabilites break down as follows:
+# Count  Probability
+# one    (1 - REPEAT_RATE[0])
+# two    REPEAT_RATE[0] * (1 - REPEAT_RATE[1])
+# three  REPEAT_RATE[0] * REPEAT_RATE[1] *(1 - REPEAT_RATE[2])
+# four   REPEAT_RATE[0] * REPEAT_RATE[1] * REPEAT_RATE[2]
+REPEAT_RATE = {1: 0.08, 2: 0.05, 3: 0.01, 4: 0.00}
+
+
+def calculate_number_of_specials_to_place(objs):
+    """Return a list of number of specials to be placed at each obj"""
+    return [1 if random.random() > REPEAT_RATE[1] else
+            2 if random.random() > REPEAT_RATE[2] else
+            3 if random.random() > REPEAT_RATE[3] else 4
+            for __ in objs]
```

Comment:
You're probably aware that in python it is convention to use an underscore to indicate that a  variable inside loops or an unpacking isn't used.  However this conflicts with another convention for the `gettext` library that uses a single underscore as alias for the `gettext` function which fetches a localized text by a unlocalized key.